src/Arcus.Messaging.ServiceBus.Core/Extensions/AzureClientFactoryBuilderExtensions.cs

```diff
@@ -51,6 +54,7 @@ public static class AzureClientFactoryBuilderExtensions
         /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="connectionStringSecretName"/> is blank.</exception>
         /// <exception cref="InvalidOperationException">Thrown when the Arcus secret store is not registered.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the registered Arcus secret store does not have a synchronous secret provider registered.</exception>
```

Comment:
I don't think we throw an exception when no sync secret provider is registered ?  We're falling back to retrieve the secret async.

Reply:
Aha, yes, removed this one and added `SecretNotFoundException`. 👍